File: src/sports_management/modules/booking_module/lib/domain/usecases/update_booking_status_usecase.dart
import 'package:server_module/server_module.dart';

class UpdateBookingStatusUseCase {
  final BookingRepository _repository;

  UpdateBookingStatusUseCase(this._repository);

  Future<BaseResponse<BookingEntity>> call(String id, String status) async {
    return await _repository.updateBookingStatus(id, status);
  }
}

File: src/react-staff-admin/src/features/booking/domain/usecases/update_booking_status.usecase.ts
import { BookingRepository } from '../repositories/booking.repository';
import { Booking } from '../entities/booking.entity';

export class UpdateBookingStatusUseCase {
  constructor(private repository: BookingRepository) {}

  async execute(id: string, status: Booking['status']): Promise<void> {
    return this.repository.updateBookingStatus(id, status);
  }
}
